arch/x86/core/fatal.c

```diff
@@ -26,6 +26,42 @@
 
 __weak void _debug_fatal_hook(const NANO_ESF *esf) { ARG_UNUSED(esf); }
 
+/**
```

Comment:
I believe all this should go into CONFIG_HW_STACK_PROTECTION, or whatever internal CONFIG you have for x86 (in ARM we have MPU_STACK_GUARD, or BUILTIN_STACK_GUARD).

At least, you somehow need to have CONFIG_THREAD_STACK_INFO defined...

Reply:
Yeah I need to fix up how this uses configs, new patch incoming

Reply:
fixed